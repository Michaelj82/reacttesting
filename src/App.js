import React from "react";
import { useState } from "react";
const App = () => {
  const [heading, setHeading] = useState("Magnificient Monkeys");

  const clickHandler = () => {
    setHeading("Radical Rhinos")
  };

  return (
    <div>
      <button type='button' onClick={clickHandler}>
        Click Me
      </button>
      <h1>{heading}</h1>
    </div>
  )
}

export default App;
