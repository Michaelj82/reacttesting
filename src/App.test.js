import React from "react";
import {render, screen} from "@testing-library/react"
import userEvent from "@testing-library/user-event";
import App from "./App";
import '@testing-library/jest-dom'
// import TestComponent from "path-to-test-component"


describe("App component", () => {
  it("renders magnificent monkeys", () =>{

    const {container} = render(<App></App>)
    expect(container).toMatchSnapshot();
  });

  it("renders radical rhinos after button clicks", () => {
    render(<App></App>);
    const button = screen.getByRole("button", {name: "Click Me"});

    userEvent.click(button);

    expect(screen.getByRole("heading").textContent).toMatch(/radical rhinos/i);

  });
});