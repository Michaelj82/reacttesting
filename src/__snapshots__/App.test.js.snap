// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`App component renders magnificent monkeys 1`] = `
<div>
  <div>
    <button
      type="button"
    >
      Click Me
    </button>
    <h1>
      Magnificient Monkeys
    </h1>
  </div>
</div>
`;
